File: motel-service/src/main/java/com/kltn/motelservice/service/NotificationService.java
package com.kltn.motelservice.service;

import com.kltn.motelservice.entity.Criteria;
import com.kltn.motelservice.entity.NotificationName;
import com.kltn.motelservice.entity.Post;
import com.kltn.motelservice.entity.User;
import com.kltn.motelservice.model.NotificationDTO;
import org.springframework.data.domain.Page;

public interface NotificationService {
    void createNotification(User user, Post post, Criteria criteria, NotificationName notificationName);

    Page<NotificationDTO> getNotificationByEmail(String email, int page, boolean screen);

    Page<NotificationDTO> getNotificationByEmailAndCriteria(String email, Long idCriteria, int page);

    NotificationDTO seenNotification(Long id);
}

File: motel-app/src/app/service/notification.service.ts
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {NotificationDTO} from '../model/notificationDTO';
import {HttpClient} from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class NotificationService {
  private apiUrl = '/api';

  constructor(private http: HttpClient) {
  }

  getNotificationByEmail(page: number, screen: boolean): Observable<NotificationDTO[]> {
    return this.http.get<NotificationDTO[]>(this.apiUrl + '/notification?page=' + page + '&screen=' + screen);
  }

  getNotificationByEmailAndCriteria(idCriteria: number, page: number): Observable<NotificationDTO[]> {
    return this.http.get<NotificationDTO[]>(this.apiUrl + '/notification/criteria?idCriteria=' + idCriteria + '&page=' + page);
  }

  seenNotification(id: number): Observable<NotificationDTO> {
    return this.http.put<NotificationDTO>(this.apiUrl + '/notification/' + id, null);
  }
}
